packaging/hudi-cli-bundle/pom.xml

```diff
@@ -0,0 +1,646 @@
+                                </transformer>
+                                <transformer implementation="org.apache.maven.plugins.shade.resource.AppendingTransformer">
+                                    <resource>META-INF/spring.handlers</resource>
+                                </transformer>
+                                <transformer implementation="org.apache.maven.plugins.shade.resource.AppendingTransformer">
+                                    <resource>META-INF/spring.schemas</resource>
+                                </transformer>
+                                <transformer implementation="org.apache.maven.plugins.shade.resource.AppendingTransformer">
+                                    <resource>META-INF/spring.factories</resource>
+                                </transformer>
+                                <transformer implementation="org.apache.maven.plugins.shade.resource.ServicesResourceTransformer"/>
+                                <transformer implementation="org.apache.maven.plugins.shade.resource.ManifestResourceTransformer">
+                                    <mainClass>org.apache.hudi.cli.Main</mainClass>
+                                </transformer>
+                                <transformer implementation="org.apache.maven.plugins.shade.resource.AppendingTransformer">
+                                    <resource>META-INF/spring/org.springframework.boot.actuate.autoconfigure.web.ManagementContextConfiguration.imports</resource>
+                                </transformer>
+                                <transformer implementation="org.apache.maven.plugins.shade.resource.AppendingTransformer">
+                                    <resource>META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports</resource>
+                                </transformer>
+                            </transformers>
+                            <artifactSet>
+                                <includes combine.children="append">
+                                    <include>org.apache.hudi:hudi-common</include>
+                                    <include>org.apache.hudi:hudi-cli</include>
+                                    <include>org.apache.hudi:hudi-client-common</include>
+                                    <include>org.apache.hudi:hudi-spark-client</include>
+                                    <include>org.apache.hudi:hudi-spark-common_${scala.binary.version}</include>
+                                    <include>org.apache.hudi:hudi-spark_${scala.binary.version}</include>
+                                    <include>org.apache.hudi:${hudi.spark.module}_${scala.binary.version}</include>
+                                    <include>org.apache.hudi:${hudi.spark.common.modules.1}</include>
+                                    <include>org.apache.hudi:${hudi.spark.common.modules.2}</include>
```

Comment:
Could we separate hudi-spark-related modules out so that the hudi-cli-bundle does not rely on Spark and Scala versions?  Right now, this can change if different Spark and Scala versions are different, and we need to provide different flavors of hudi-cli-bundles, which can be hard to maintain.

If we separate out the Hudi Spark related modules, we can ask the use to the hudi-cli-bundle with the hudi-spark-bundle corresponding to the Spark version used.

Reply:
Makes sense.